### DISCUSSION TITLE
New Twitter Auth Bugs and Outdated Documentation - Help Needed

### DISCUSSION THREAD

**Discussion by ethanchiasson** [supabase/supabase#8996] [Questions]
I made an issue for this but thought I would make a discussion around it.

At the time of writing, it seems as though Supabase's Twitter Auth functionality is broken. For more information on the issue, please check the issue linked below.

https://github.com/supabase/auth-helpers/issues/246

This bug - that is, getting an error `http://localhost:3000/?error=server_error&error_description=Error+getting+user+email+from+external+provider` - has been brought up in the past, but each time there seems to be a different cause ( and fix ).

Here is a quick outline of currently known or previously known Twitter Authentication bugs / issues.

1. Twitter accounts with no emails can't pass auth since Supabase needs an email for its Auth flow - https://github.com/supabase/supabase/issues/2853
https://github.com/supabase/gotrue/issues/214

2. There was an issue where a user needed to upgrade their Twitter Project to `elevated` status within the Twitter Developer Portal to be able to read the email from a user's account ( see #1 ). 

**This has still yet to be added to the Supabase documentation here : https://supabase.com/docs/guides/auth/auth-twitter
https://github.com/supabase/auth-helpers/issues/100
https://github.com/supabase/supabase/discussions/4677
https://github.com/supabase/supabase/issues/8127

3. The *New Issue* is after Obtaining Elevated access on my Twitter Developer account, I am still unable to authenticate the user, getting the same URL embedded error message as previously thought to be fixed . 
https://github.com/supabase/auth-helpers/issues/246

I do not believe issue 3 is a setup issue, since other user's seem to be having the same issue.

I just wanted to make this discussion thread in hopes that...

1. This post can help other users who are stuck on this issue have a resource to understand the timeline of events.
2. Get attention from the Supabase team to fix the issue ( if there is one ) and update the documentation to include the Twitter Elevated Project requirements.

Thanks for reading.

**Comment by mikelxc** [comment]
Still getting the {"message":"No API key found in request"} when using Twitter OAuth API, did some debugging and I think it's related to Kong sending this error message

**Reply by mikelxc** [reply]
And for the normal consumer api, the login does work, but it cannot return a email address, resulting an error

**Reply by ethanchiasson** [reply]
did you ever find a fix ?

**Reply by mistakenot** [reply]
I fixed this by setting twitter to return the email address for the user. 

<img width="649" alt="image" src="https://user-images.githubusercontent.com/7852099/193343848-47c8e78c-732e-4048-a297-8b3c89fdb576.png">

Let me know if this works.

**Reply by ethanchiasson** [reply]
I have this activated already so I do not think this is the issue.
